include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -826,21 +857,26 @@ template <typename _BackendTag, typename _ExecutionPolicy, typename _Range, type
 oneapi::dpl::__internal::__difference_t<_Range>
 __pattern_unique(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Range&& __rng, _BinaryPredicate __pred)
 {
-    if (__rng.size() == 0)
-        return __rng.size();
+    const auto __n = oneapi::dpl::__ranges::__size(__rng);
+    if (__n == 0)
+        return __n;
 
     using _ValueType = oneapi::dpl::__internal::__value_t<_Range>;
 
-    oneapi::dpl::__par_backend_hetero::__buffer<_ValueType> __buf(__rng.size());
-    auto res_rng = oneapi::dpl::__ranges::views::all(__buf.get_buffer());
+    auto&& __rng_s = oneapi::dpl::__ranges::__get_subscription_view(__rng);
```

Comment:
Should it rather be:
```suggestion
    auto __rng_s = oneapi::dpl::__ranges::__get_subscription_view(std::forward<_Range>(__rng));
```

Reply:
Why `auto&& __rng_s` steel needed: because it's preserve value category of returned value.
This allows us to write
```C++
std::forward<decltype(__rng_s)>(__rng_s)
```
later in the code and it will make sense: if `__get_subscription_view` returns source parameter `__rng` without changes, we will preserve their value category.